.github/workflows/test-deploy.yml

```diff
@@ -4,23 +4,31 @@ on:
     - cron: 0 0 * * *
   release:
     types: [published]
-  pull_request:
+  pull_request_target:
 jobs:
+  authorize:
+    environment:
+      ${{ (github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository) &&
+      'external' || 'internal' }}
+    runs-on: ubuntu-latest
   lint:
+    needs: authorize
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.ref }}
```

Comment:
why is this required?

Reply:
Because `pull_request_target` would checkout the target branch by default, not the source branch.

Reply:
don't get it. Are you saying `actions/checkout@v2` needs configuring for `on.*` (including specifically config for `ref` for `on.pull_request_target`)? I doubt it.

Reply:
After some reading it looks like `actions/checkout` *is* indeed broken. Need to do some more reading before I'd be happy to accept this. Grrr, GitHub.

Reply:
Not actually broken: https://securitylab.github.com/research/github-actions-preventing-pwn-requests

Reply:
It just uses the target branch instead of the source branch to avoid running privileged code from untrusted sources.

Reply:
> Not actually broken: [securitylab.github.com/research/github-actions-preventing-pwn-requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests)

that link is amongst many things the precise definition of broken :)

Reply:
> _that link is amongst many things the precise definition of broken_

If you mean literally broken... it works on my computer. ™ 🤔  Please try [reading it from Archive.org]( https://web.archive.org/web/20210426102036/https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)